modifier.go

```diff
@@ -2,83 +2,100 @@ package main
+	flag.StringVar(&configPath, "config", "./config.yml", "Path to configuration file.")
+	flag.DurationVar(&duration, "duration", 0, "Time modify duration. Must be negative for now.")
+	flag.StringVar(&inputFile, "i", "./input/Hz.Omer.S01E01.srt", "Path to input file.")
+	flag.StringVar(&outputFile, "o", "./output/Hz.Omer.S01E01.srt", "Path to output file.")
+	flag.Parse()
+
+	if inputFile != "" && outputFile != "" && duration < 0 {
+		err := process(inputFile, outputFile, duration)
+		if err != nil {
+			log.Fatalf("could not process file %s: %v", inputFile, err)
+		}
+		os.Exit(0)
 	}
 
-	inputFile := os.Args[1]
-	outputFile := os.Args[2]
-	modifyValue, _ := strconv.Atoi(os.Args[3])
+	configuration, err := config.GetConfiguration(configPath)
+	if err != nil {
+		log.Fatalf("could not load config from path %s: %v\n", configPath, err)
+	}
 
-	fmt.Printf("Processing %v\n", inputFile)
+	var wg sync.WaitGroup
+	for _, cf := range configuration.Files {
+		wg.Add(1)
+		go func(iFile string, oFile string, mVal time.Duration) {
+			defer wg.Done()
+			err := process(iFile, oFile, mVal)
+			if err != nil {
+				log.Printf("could not process file %s: %v\n", iFile, err)
+			}
+		}(cf.InputFile, cf.OutputFile, cf.ModifyValue)
+	}
+	wg.Wait()
+}
 
+func process(inputFile string, outputFile string, modifyValue time.Duration) error {
 	inpFile, err := os.Open(inputFile)
-
 	if err != nil {
-		log.Fatal(err)
-		return
+		return errors.Wrap(err, "could not open input file")
 	}
+	defer inpFile.Close()
 
-	fmt.Printf("Creating %v if not exists\n", outputFile)
 	outFile, err := os.Create(outputFile)
-
 	if err != nil {
-		log.Fatal(err)
-		return
+		return errors.Wrap(err, "could not create output file")
 	}
-
-	defer inpFile.Close()
 	defer outFile.Close()
 
 	scanner := bufio.NewScanner(inpFile)
-
-	subtitleCounter := 0
-	regexPattern, _ := regexp.Compile("[0-9]{2}:[0-9]{2}:[0-9]{2},[0-9]{3} --> [0-9]{2}:[0-9]{2}:[0-9]{2},[0-9]{3}")
-
 	for scanner.Scan() {
-
 		text := scanner.Text()
 
 		if regexPattern.MatchString(text) {
-			// split the time into start time and end time
 			times := strings.Split(text, " ")
 			startTime := times[0]
 			endTime := times[2]
 
-			// subtract time from start time
 			modifiedStartTime := addValueToTime(startTime, modifyValue)
-
-			// subtract time from end time
 			modifiedEndTime := addValueToTime(endTime, modifyValue)
-
-			// combines the time
-			text = modifiedStartTime + " --> " + modifiedEndTime
+			text = fmt.Sprintf("%s --> %s", modifiedStartTime, modifiedEndTime)
 		}
-		// write processed information into file
 		outFile.WriteString(text + "\n")
-		subtitleCounter += 1
 	}
 
 	fmt.Printf("Processed %v\n", inputFile)
 	if err := scanner.Err(); err != nil {
-		log.Fatal(err)
+		return errors.Wrap(err, "could not scan")
 	}
 
+	return nil
+
 }
 
 // currently supports subtraction of time
-func addValueToTime(time string, value int) string {
+func addValueToTime(time string, value time.Duration) string {
```

Comment:
[optional] you plan to also support add in the future so using add/subtract in the name could be misleading:
```suggestion
func applyOffset(time string, offset time.Duration) string {
```